### PR TITLE
Support other layer types beyond Labels 

### PR DESCRIPTION
**Pull request by haesleinhuepf**
This is an attempt to plot properties of surface layers. I presume it is at hand to implement the same for points layers ( #103 ).

To try it out, you need [napari-process-points-and-surfaces]() and in particular the menu `Tools > Measurements > Surface Quality Table (nppas)` to put measurements in `surface_layer.properties`.

I managed to make it work for Dimensionality Reduction, Clustering and Plotting. One can for example cluster the Stanford Rabbit's vertices like this:
image

I'm afraid the right `color` is not yet supported by napari. I will create an issue there as well.

I presume the same should be possible for Points layers as well.

Please don't merge this yet. We should add tests, example data, documentation and potentially support for Points layers first. Also #117 should be merged before.

### CODE REVIEW

**Comment by lazigu** [comment]
Hi Robert @haesleinhuepf,
This looks amazing! 
From where could I get this bunny's dataset to play around with?

**Comment by haesleinhuepf** [comment]
It's part of napari-process-points-and-surfaces. There you also find the tabular measurements needed for clustering:
https://github.com/haesleinhuepf/napari-process-points-and-surfaces#surface-measurements-and-annotations

File: napari_clusters_plotter/_utilities.py [hunk 1]
```diff
@@ -102,27 +103,50 @@ def update_properties_list(widget, exclude_list):
...
+        gpu_labelimage = cle.push(label_image)
+
+        # generation of cluster label image
+        parametric_image = cle.replace_intensities(gpu_labelimage, clelist)
+        gpu_labelimage = None
+        clelist = None
+
+        # retrieving the gpu image
+        output = cle.pull(parametric_image).astype("uint32")
+        parametric_image = None
```
Comment (lazigu) [inline napari_clusters_plotter/_utilities.py:140]: I think [Relabel Image - change how the measurements list is reformed #34](https://github.com/haesleinhuepf/napari-skimage-regionprops/pull/34) in napari-skimage-regionprops should be merged, and then the following code changed to:
```suggestion
        output = relabel(label_image, list(predictionlist_new)).astype("uint32")
```

**Comment by lazigu** [comment]
Hi @haesleinhuepf,
Do you mind if I take over this PR and directly commit to this branch? 
I will work on this to-do list then:

- [ ] fix existing tests
- [ ] add more tests
- [ ] add example data
- [ ] add documentation
- [ ] support for points layers

Should example data be the same as in https://github.com/haesleinhuepf/napari-process-points-and-surfaces or do you have something else in mind?

**Comment by haesleinhuepf** [comment]
Hi Laura! Oh it would be awesome if you could take this task over. @Cryaaa will send us cool example data soon ;-)

**Comment by Cryaaa** [comment]
@haesleinhuepf @lazigu 
Yep, here is some organoid data which you can use for surface reconstruction:
https://zenodo.org/record/7603081

The images are quite huge so you might want to do some down-scaling and smoothing!

**Comment by lazigu** [comment]
@Cryaaa, thank you so much for this!

File: napari_clusters_plotter/_utilities.py (hunk 1 above)
Comment (haesleinhuepf) [reply napari_clusters_plotter/_utilities.py]: Done! :-)

**Comment by haesleinhuepf** [comment]
Ok, [here](https://github.com/haesleinhuepf/napari-process-points-and-surfaces/blob/9f80178732aa1e33d11d960eadb34a9d75cbbb72/docs/gastruloid.ply) comes a ply (mesh/surface) file of a Gastruloid. When using it, please put this statement in the code or notebook:

```
The used example data is derived from of [AV Luque and JV Veenvliet (2023)](https://zenodo.org/record/7603081#.Y94N4XbMJmN) licensed [CC-BY](https://creativecommons.org/licenses/by/4.0/legalcode).
```

You can open it in napari using [napari-stl-exporter](https://github.com/jo-mueller/napari-stl-exporter) and from Python as shown [here](https://github.com/haesleinhuepf/napari-process-points-and-surfaces/blob/9f80178732aa1e33d11d960eadb34a9d75cbbb72/docs/visualizing_surfaces.ipynb):

```
import napari_process_points_and_surfaces as nppas
import vedo
import napari

mesh = vedo.load("gastruloid.ply")
surface = nppas.to_napari_surface_data(mesh)

viewer = napari.Viewer(ndisplay=3)
viewer.add_surface(surface)
```

**Comment by lazigu** [comment]
Hi, Marcelo @zoccoler,

I fixed the bug I was talking about yesterday and other merge conflicts and this branch can safely be used as a starting point today. One thing is that Dask is not used for cluster image generation with surface layers. How frequent are surface timelapses anyway? We don't have any surface example data that is 4D anyway for testing.

Other todo points that were planned for this branch (copied from my old comment above):

- [x] fix existing tests
- [ ] add surfaces specific tests
- [ ] add example data (see few messages above for the example data)
- [ ] use Dask for surface timelapse data?
- [ ] custom colour maps do not work for surfaces (see linked issues above)
- [ ] add documentation
- [ ] support for points layers

**Pull request closed without merging**